bundles/core/org.eclipse.smarthome.core.thing.test/src/test/java/org/eclipse/smarthome/core/thing/firmware/FirmwareUpdateServicePrerequisiteVersionTest.java

```diff
@@ -0,0 +1,121 @@
+        registerService(createFirmwareProvider(thingWithFirmwareV2, firmwareV1, firmwareV2, firmwareV3));
+        registerService(createFirmwareUpdateHandler(thingWithFirmwareV1));
+        registerService(createFirmwareUpdateHandler(thingWithFirmwareV2));
+    }
+
+    @SuppressWarnings("null")
+    @Test
+    public void testGetFirmwareStatusInfoWithUnsatisfiedPrerequisiteVersion() {
+        FirmwareUpdateService firmwareUpdateService = getService(FirmwareUpdateService.class);
+
+        FirmwareStatusInfo firmwareStatusInfo = firmwareUpdateService
+                .getFirmwareStatusInfo(thingWithFirmwareV1.getUID());
+        assertThat(firmwareStatusInfo, notNullValue());
+
+        // Since firmware version 3.0.0 has prerequisite version 2.0.0, and the thing has firmware version 1.0.0, the
+        // expectation is that the version of the firmware to which one can update is 2.0.0.
+        assertThat(firmwareStatusInfo.getUpdatableFirmwareVersion(), equalTo("2.0.0"));
+    }
+
+    @SuppressWarnings("null")
+    @Test
+    public void testGetFirmwareStatusInfoWithSatisfiedPrerequisiteVersion() {
+        FirmwareUpdateService firmwareUpdateService = getService(FirmwareUpdateService.class);
+
+        FirmwareStatusInfo firmwareStatusInfo = firmwareUpdateService
+                .getFirmwareStatusInfo(thingWithFirmwareV2.getUID());
+        assertThat(firmwareStatusInfo, notNullValue());
+
+        // Since firmware version 3.0.0 has prerequisite version 2.0.0, and the thing has firmware version 2.0.0, the
+        // expectation is that the version of the firmware to which one can update is 3.0.0.
+        assertThat(firmwareStatusInfo.getUpdatableFirmwareVersion(), equalTo("3.0.0"));
+    }
+
+    private Thing createThing(ThingTypeUID thingTypeUID, String thingUID, Firmware firmware) {
+        Thing thing = mock(Thing.class);
+        when(thing.getThingTypeUID()).thenReturn(thingTypeUID);
+        when(thing.getUID()).thenReturn(new ThingUID(thingTypeUID, thingUID));
+        when(thing.getProperties()).thenReturn(ImmutableMap.<String, String> builder()
+                .put(Thing.PROPERTY_FIRMWARE_VERSION, firmware.getVersion()).build());
+        return thing;
+    }
+
+    private FirmwareProvider createFirmwareProvider(Thing thing, Firmware... firmwares) {
+        FirmwareProvider firmwareProvider = mock(FirmwareProvider.class);
+        when(firmwareProvider.getFirmwares(eq(thing), any(Locale.class)))
+                .thenReturn(new HashSet<>(Arrays.asList(firmwares)));
+        return firmwareProvider;
+    }
+
+    private FirmwareUpdateHandler createFirmwareUpdateHandler(Thing thing) {
+        FirmwareUpdateHandler firmwareUpdateHandler = mock(FirmwareUpdateHandler.class);
+        when(firmwareUpdateHandler.getThing()).thenReturn(thing);
+        when(firmwareUpdateHandler.isUpdateExecutable()).thenReturn(true);
+        return firmwareUpdateHandler;
+    }
+
+}
```

Comment:
Please add a line break after the last line

Reply:
Thanks, will do (and should have looked at the static code analysis one more time before creating this PR...).